lib/scholar/linear/isotonic_regression.ex

```diff
@@ -495,4 +505,11 @@ defmodule Scholar.Linear.IsotonicRegression do
 
     if opts[:increasing?], do: y, else: Nx.reverse(y)
   end
+
+  deftransform check_increasing(x, y) do
+    x = Nx.new_axis(x, -1)
+    y = Nx.new_axis(y, -1)
+    model = Scholar.Linear.LinearRegression.fit(x, y)
+    Nx.greater_equal(model.coefficients[0][0], 0) == Nx.u8(1)
```

Comment:
This is not going to work if you call isotonic_regression from within a defn, because all you have is expressions, never values. You can try by doing this:

```elixir
defn foo do
  call_isotonic_stuff(...)
end
```

It will always be false. I think what you will have to do is to keep this as `defnp`. Then increasing? should either be 0, 1, or the result of this expression, and you pass increasing as a tensor to fit_n, rather than an option.

Reply:
We need to remove the comparison from here:

```suggestion
    Nx.greater_equal(model.coefficients[0][0], 0)
```